### ISSUE TITLE
Discuss: separating Host configuration from Engine

### ISSUE THREAD

**Issue by obuchtala** [stencila/engine#3]
At the moment, we have a smart Host that is IMO overly complicated.
In the current demo in this repo: https://github.com/stencila/engine/blob/master/example/run.js#L9-L18
I was playing with the idea, that there is a configuration phase, setting up proxies, instead of tangling the evaluation with this setup work.
Of course the Host could be reconfigured at any time.

**Comment by obuchtala** [comment]
@nokome maybe we can discuss this too

**Comment by nokome** [comment]
Yes, I was thinking similarly. The concept of a client side `Host` has always been awkward/confusing. There is some complexity in there that is historical artefact and could be removed.

But we also need to be careful about throwing out the baby with the bathwater - still important to have dynamic reconfiguration, looking up which host has which contexts available when a new type of context is requested by the user.
